src/main/kotlin/CheckersModel.kt

```diff
@@ -0,0 +1,113 @@
+                return Pair(false, null)
+            }
+            val verticals = turn.to.second - turn.from.second
+            val horizontals = turn.to.first - turn.from.first
+            if ((horizontals == if (turn.playerColor == Color.WHITE) 1 else -1) && (abs(verticals) == 1)) {
+                return Pair(true, null)
+            }
+            if ((abs(horizontals) == 2) && (abs(verticals) == 2)
+                    && (board[turn.from.first][turn.from.second].figure!!.type == FigureType.Ordinary)) { //Если просто съедаем через клетку
+                val squareToEat = board[(turn.from.first + turn.to.first) / 2][(turn.from.second + turn.to.second) / 2]
+                if ((squareToEat.figure != null) && (squareToEat.figure?.color != squareFrom.figure!!.color)) { //проверяем что там стоит шашка другого цвета
+                    return Pair(true, squareToEat)
+                }
+            }
+            /*if ((abs(horizontals) == abs(verticals)) && (squareFrom.figure!!.type == FigureType.Queen)) { //съедаем через несколько клеток(должна быть дамкой)
+                var count: Int = 0
+                for (i in 1 until abs(horizontals) + 1) {
+
+                }
+            }*/
+            return Pair(false, null)
+        } else {
+            //клетка "откуда" пустая
+            return Pair(false, null)
+        }
+    }
+
+    override fun move(turn: BaseTurn) {
+        val canMoveResult = canMove(turn)
+        if (!canMove(turn).first)
+            return
+        run {
+            board[turn.to.first][turn.to.second].figure = board[turn.from.first][turn.from.second].figure
+            board[turn.from.first][turn.from.second].figure = null
+            whoMoves = if (whoMoves == Color.WHITE)
+                Color.BLACK
+            else
+                Color.WHITE
+            updateState()
+        }
+        if (canMoveResult.second != null) {
+            (canMoveResult.second as Square).figure = null
+        }
+
+        //TODO("Сделать съедание шашек")
+        //TODO("Сделать превращение в дамку")
+    }
+
+    override fun updateState() {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun printBoardOnConsole() {
+        for (i in boardSize - 1 downTo 0) {
+            for (j in 0 until boardSize) {
+                var t = "."
+                if (board[i][j].figure == null) {
+                    t = "."
+                } else {
+                    if (board[i][j].figure!!.color == Color.WHITE) {
+                        t = "w"
+                    } else {
+                        t = "b"
+                    }
+                    if (board[i][j].figure!!.type == FigureType.Queen) {
+                        t.toUpperCase()
+                    }
+                }
+                print(t)
+            }
+            println();
+        }
+    }
+
+    init {
```

Comment:
Should go to the init block of Board class. By the way, init blocks are usually placed in the beginning of the class - in fact, they are class constructors.